core/src/main/java/org/infinispan/commands/write/IracPutKeyValueCommand.java

```diff
@@ -174,9 +176,18 @@ public void readFrom(ObjectInput input) throws IOException, ClassNotFoundExcepti
       commandInvocationId = CommandInvocationId.readFrom(input);
       privateMetadata = (PrivateMetadata) input.readObject();
       segment = UnsignedNumeric.readUnsignedInt(input);
+      expiration = input.readBoolean();
       setFlagsBitSet(FlagBitSets.IRAC_UPDATE);
    }
 
+   public boolean isExpiration() {
+      return expiration;
+   }
+
+   public void setExpiration(boolean expiration) {
```

Comment:
Not sure why we would want to know if during a put the entry was written was also expired with irac? I thought we would just ignore expiration completely during an irac write?

Reply:
Or is this just for a tombstone?

Reply:
The name of the class is not the best since it is used for writes and removes from remote site :/

it is just to signal that this remove operation is from expiration and it should lose in case of conflicts. See https://github.com/infinispan/infinispan/pull/9346/files#diff-ffada9becbfda8ba5b90e9f43487b4508ce0f8f3d644547104eb0ae5823beab3R153